module3/src/com/epam/stringorstringBuilder/StringWorker.java

```diff
@@ -0,0 +1,152 @@
+package com.epam.stringorstringBuilder;
+
+
+public class StringWorker {
+    //task 1
+    public static int calculateLargestNumberOfSpaces(String text) {
+        int lengthOfSpaces = 0;
+        int maxLengthOfSpaces = 0;
+        final char[] characters = text.toCharArray();
+        for (int i = 0; i < characters.length; i++) {
+            if (characters[i] == ' ') {
+                lengthOfSpaces++;
+                if (lengthOfSpaces > maxLengthOfSpaces) {
+                    maxLengthOfSpaces = lengthOfSpaces;
+                }
+            } else lengthOfSpaces = 0;
+        }
+        return maxLengthOfSpaces;
+    }
+
+    //task 2
+    public static String insertBAfterAch(String text) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < text.length(); i++) {
+            if (text.charAt(i) == 'a') {
+                stringBuilder.append("ab");
+            } else if ((text.charAt(i) == 'A')) {
+                stringBuilder.append("AB");
+            } else {
+                stringBuilder.append(text.charAt(i));
+            }
+        }
+        return stringBuilder.toString();
+    }
+
+    //task 3
+    public static boolean isPalindrome(String word) {
+        int wordLength = word.length();
+        for (int i = 0; i < (wordLength / 2); i++) {
+            if (word.charAt(i) != word.charAt(wordLength - i - 1)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    //task 4
+    public static String convertInfomaticaToCake(String word) {
+        String cake = String.valueOf(word.charAt(word.indexOf('т')));
+        cake += String.valueOf(word.charAt(word.indexOf('о')));
+        cake += String.valueOf(word.charAt(word.indexOf('р')));
+        cake += String.valueOf(word.charAt(word.indexOf('т')));
+        return cake;
+    }
+
+    //task 5
+    public static int calculateTheA(String text) {
+        int count = 0;
+        for (int i = 0; i < text.length(); i++) {
+            if (text.charAt(i) == 'a' || (text.charAt(i) == 'A')) {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    //task 6
+    public static String charactersDoubler(String text) {
+        StringBuilder stringBuilder = new StringBuilder();
+        int numberOfRepetitions = 2;
+        for (int i = 0; i < text.length(); i++) {
+            for (int j = 0; j < numberOfRepetitions; j++) {
+                stringBuilder.append(text.charAt(i));
+            }
+        }
+        return stringBuilder.toString();
+    }
+
+    //task 7
+    public static String deleteRepeatedCharactersAndSpaces(String text) {
+        String textWithOutSpaces = deleteSpaces(text);
+        return deleteRepeatedCharacters(textWithOutSpaces);
```

Comment:
I can provide you the solution which is shorter than yours:
```suggestion
        char[] characters = str.toCharArray();
        StringBuilder modifiedStr = new StringBuilder();
        for (char ch : characters) {
            if (ch != ' ' && modifiedStr.indexOf(String.valueOf(ch)) == -1) {
                modifiedStr.append(ch);
            }
        }
        return modifiedStr.toString();
```